### PR TITLE
cleanup blackbox tests

### PR DESCRIPTION
**Pull request by ajeddeloh**
Fix a few blackbox test bugs and do some cleanup and refactoring to make future work easier.

### CODE REVIEW

File: tests/filesystem.go [hunk 1]
```diff
@@ -127,42 +127,60 @@ func pickPartition(t *testing.T, device string, partitions []*types.Partition, l
...
 	// Truncate the file to the given size
-	err = os.Truncate(imageFile, size)
-	if err != nil {
+	if err = os.Truncate(disk.ImageFile, imageSize); err != nil {
 		return err
 	}
 
-	err = createPartitionTable(t, ctx, imageFile, partitions)
+	// Attach the file to a loopback device
+	tmp, err = run(t, ctx, "losetup", "-Pf", "--show", "-b", "512", disk.ImageFile)
```
Comment (arithx) [inline tests/filesystem.go]: Should sector size be added to `disk` and referenced here rather than the hardcoded 512?

Comment (ajeddeloh) [reply tests/filesystem.go]: oh whoops the `-b` was meant to be left off; I added this while debugging something. I probably going to add another attribute to `tests/types.Disk` to specify block size later. I'll remove this.

Comment (ajeddeloh) [reply tests/filesystem.go]: fixed

File: tests/positive/timeouts/timeouts.go [hunk 2]
```diff
@@ -36,15 +36,16 @@ var (
 		w.WriteHeader(http.StatusOK)
 	}))
 
-	lastResponse           time.Time
+	lastResponses          = map[string]time.Time{}
```
Comment (arithx) [inline tests/positive/timeouts/timeouts.go]: Nice solution!

File: tests/blackbox_test.go [hunk 3]
```diff
@@ -136,54 +136,48 @@ func outer(t *testing.T, test types.Test, negativeTests bool) error {
...
-		// Move value into the local scope, because disk.ImageFile will change
-		// by the time this runs
+		// Move value into the local scope, because disk.ImageFile and device
+		// will change by the time this runs
 		imageFile := disk.ImageFile
+		device := disk.Device
 		defer func() {
-			err := os.Remove(imageFile)
-			if err != nil {
+			t.Logf("removing file")
```
Comment (arithx) [inline tests/blackbox_test.go]: Is this additional log output accidental?

Comment (ajeddeloh) [reply tests/blackbox_test.go]: Accidental

**Comment by arithx** [comment]
New changes LGTM, can you repush with a fix for https://github.com/coreos/ignition/pull/616#discussion_r211025861 and then good to merge.

**Pull request merged**